### PR TITLE
Vue3 support - NPM Package in comments!

### PR DESCRIPTION
**Pull request by james090500**
This build supports vue3 with basic functionality. Still some basic errors which need resolving but it at least works on vue3 for now. 

This is a temporary build until we can get something final together.

### CODE REVIEW

**Comment by james090500** [comment]
I fixed some render code and broken version code. This runs pretty optimal now to get you guys on vue3!

You can install with: ``npm i https://github.com/james090500/vue-stripe#james090500/vue3``

**IMPORTANT**
You MUST put the slot inside the element. 
These are breaking changes which confused me for a bit:

Vue 2:
```
<stripe-checkout :ref="this.type" :pk="this.pk" :session-id="this.sessionId"/>
<button class="btn btn-primary">Pay Now</button>
```

Vue 3:
```
<stripe-checkout :ref="this.type" :pk="this.pk" :session-id="this.sessionId">
      <button class="btn btn-primary">Pay Now</button>
 </stripe-checkout>
```

**Comment by jofftiquez** [comment]
Thank you so much for this @james090500 really helpful while the official one is still in development. I have created a list of update in the vue 3 issue discussion. I just wanna say that I cannot merge this PR anymore as there are core changes to the version 5 and it's just a full rehaul really. Still thank you, I will keep this PR open until we finalized the new version. The vue 2 version is still going to be maintained for a while so if you have time any type of help would be appreciated. Thank you again.

File: README.md
Comment (jofftiquez) [inline README.md:1]: Hi no need need to update the username in these links thanks

File: .vscode/extensions.json
Comment (jofftiquez) [inline .vscode/extensions.json:1]: Please remove

File: .eslintrc.cjs
Comment (jofftiquez) [inline .eslintrc.cjs:1]: Do not replace the original eslitrc file as it breaks all the intentional rules I have set to the project

File: babel.config.js
Comment (jofftiquez) [inline babel.config.js:1]: Why remove?

File: index.html
Comment (jofftiquez) [inline index.html:1]: Hi, what's the purpose of this? Thanks

File: package-lock.json
Comment (jofftiquez) [inline package-lock.json:1]: Please remove, I am using yarn exclusively

File: package.json [hunk 1]
```diff
@@ -3,59 +3,42 @@
...
   },
   "bugs": {
-    "url": "https://github.com/vue-stripe/vue-stripe/issues"
-  },
-  "gitHooks": {
-    "pre-commit": "lint-staged"
+    "url": "https://github.com/james090500/vue-stripe/issues"
   },
-  "homepage": "https://github.com/vue-stripe/vue-stripe#readme",
+  "homepage": "https://github.com/james090500/vue-stripe#readme",
```
Comment (jofftiquez) [inline package.json:41]: Revert to the original

File: package.json [hunk 2]
```diff
@@ -3,59 +3,42 @@
...
+    "vite": "^4.0.4",
+    "vite-plugin-html": "^3.2.0",
+    "vue-tsc": "^1.0.22"
   },
   "bugs": {
-    "url": "https://github.com/vue-stripe/vue-stripe/issues"
-  },
-  "gitHooks": {
-    "pre-commit": "lint-staged"
+    "url": "https://github.com/james090500/vue-stripe/issues"
```
Comment (jofftiquez) [inline package.json:39]: Revert to the original

File: package.json [hunk 3]
```diff
@@ -64,15 +47,8 @@
...
-      "vue-cli-service lint",
-      "git add"
-    ]
-  },
   "repository": {
     "type": "git",
-    "url": "git@github.com:vue-stripe/vue-stripe.git"
-  },
-  "typings": "typings/index.d.ts"
+    "url": "git@github.com:james090500/vue-stripe.git"
```
Comment (jofftiquez) [inline package.json:52]: Revert to the original url

File: src/checkout/index.js
Comment (jofftiquez) [inline src/checkout/index.js:1]: Fix lint formatting, don't use double quotes

File: src/elements/Payment.vue [hunk 4]
```diff
@@ -2,29 +2,22 @@
...
-          id="stripe-payment-element-errors"
-          role="alert"
-        />
-      </slot>
-      <button
-        ref="submitButtonRef"
-        type="submit"
-        class="hide"
-      />
+      <mitB name="stripe-payment-element-errors">
```
Comment (jofftiquez) [inline src/elements/Payment.vue:5]: What is `mitB` for?

**Review by jofftiquez — CHANGES_REQUESTED** [review]

**Comment by jofftiquez** [comment]
Closing due to lack of response to the code review. Thank you

**Comment by james090500** [comment]
Hi,

Sorry for the lack of response, this must have slipped through my emails. My changes to the username were so I could push a copy to npm for my code, I've realised better ways of doing this since then.

Hopefully my code was somewhat helpful anyway.

**Comment by jofftiquez** [comment]
> Hi,
> 
> Sorry for the lack of response, this must have slipped through my emails. My changes to the username were so I could push a copy to npm for my code, I've realised better ways of doing this since then.
> 
> Hopefully my code was somewhat helpful anyway.

Hi, is surely did give me some ideas. Thank you so much.

**Comment by james090500** [comment]
No worries. Glad to have provided a temporary stopgap.

**Pull request closed without merging**